web/packages/shared/utils/highbar.ts

```diff
@@ -68,33 +68,51 @@ export function arrayObjectIsEqual(arr1: unknown[], arr2: unknown[]): boolean {
   );
 }
 
-/** Recursively compares two values. */
-export function equalsDeep(val1: unknown, val2: unknown) {
-  if (!isObject(val1)) {
+/**
+ * Recursively compares two values. If `ignoreUndefined` is set to `true`,
+ * treats fields set to `undefined` as equal to fields that don't exist at all.
+ */
+export function equalsDeep(
+  val1: unknown,
+  val2: unknown,
+  ignoreUndefined?: boolean
+) {
+  if (!isObject(val1) || !isObject(val2)) {
     return val1 === val2;
   }
 
   if (Array.isArray(val1) && Array.isArray(val2)) {
     return arrayObjectIsEqual(val1, val2);
```

Comment:
Should we pass `ignoreUndefined` here as well?

Reply:
🤦 You're absolutely right.